### DISCUSSION TITLE
State of Wegue v2 Pull Requests

### DISCUSSION THREAD

**Discussion by chrismayer** [meggsimum/wegue#326] [General]
Hi all, there is a lot of things going on which pay in for Wegue v2, which is really cool. Thanks to all, who are involved :pray: 

Especially the PRs #314 and #320 seem to be the major parts and a lot of technical details are shared within the PR discussions (which is good!). What I am curious about is how the people involved (mainly @fschmenger @spwoodcock @sronveaux) would judge the state of those in sense of a higher perspective. Are those PRs nearly done and are close to be merged or do they need way more work? If one are not technically involved in deep way it is hard to judge how far things are currently.

Thanks for any feedback in advance.

**Comment by fschmenger** [comment]
Hi Christian,
I think we are close to finalizing both #314 and #320. Maybe it could be useful to open another discussion topic regarding the Wegue Roadmap. So we could talk about which of the open issues should be addressed before moving to v2 and which we might handle later on?!

**Reply by chrismayer** [reply]
https://github.com/wegue-oss/wegue/discussions/327

**Comment by sronveaux** [comment]
Hi Christian,
I also personally think we are really close to it. #314 just needs to be amended with known patches and #320 seems to have two constants to be modified inside unit tests to be ready.
I can only give feedback based on a single `Wegue` app by the way (and the `app-starter` one of course!) but @fschmenger seems to have tested those two PRs with more apps with positive results which is really great!

**Reply by sronveaux** [reply]
Concerning #320 I just forgot there was another potential thing here as the unit tests can't be run inside a `Docker` image for now. No more discussion here as I think they should take place inside #320...

**Reply by spwoodcock** [reply]
The Docker issue I mentioned was specific to my setup, as my dev environment is actually inside a Docker container (remote). So I couldn't run the Karma tests which rely on Chrome. 

I don't think there is a requirement for the tests to be run in a container though.

**Reply by sronveaux** [reply]
Hi @spwoodcock ,
O.K., I wrote it here as you mentioned it but I fully agree with you, I also think unit tests shouldn't be run in a container.
We just have to keep it in mind if `Wegue` includes some e2e tests in the future as those should be able to be run against a container. However these should run against the container and not inside so this particular point shouldn't be a problem.

**Comment by chrismayer** [comment]
Thanks for your feedback @fschmenger and @sronveaux. This sound quite promising. I also did just some test regarding #320 and it works really well.

To give some background why I asked how far we are: The today master has currently a lot of vulnerabilities and the ongoing PRs will drastically reduce them. So it would be really cool to have these change in quite soon. Also starting a new project with a lot of vulnerabilities might scare people off. And starting based on the master branch is IMHO less a boundary than starting your project on an ongoing PR.

Even it might sound like this I my intention is **not** to create pressure, since I know we are all doing this in our spare time or as work side projects.

**Reply by fschmenger** [reply]
As far as I am concerned, we could merge #325. Then we are good to go to split up the 2 repos as proposed in #321.

**Reply by chrismayer** [reply]
By 2 repos you mean 2 branches, right?

**Reply by sronveaux** [reply]
Agree 100% with all what is written up here. #314 can be ready next week I think and #320 can certainly follow shortly.
The latter doesn't impact the vulnerabilities count as much as #314 so can, I think, be delayed a little bit if needed...

**Reply by fschmenger** [reply]
Not sure what the best practice is. Using branches apparently has the disadvantage, that a new user might have to switch to the `stable` branch first. Here`s is some entrypoint at [Stack overflow](https://stackoverflow.com/questions/7362825/multiple-branch-vs-multiple-repo), but I have no time for deeper investigations currently.  I leave the decision to you.

**Comment by chrismayer** [comment]
As suggested by @fschmenger I created a roadmap discussion: https://github.com/wegue-oss/wegue/discussions/327 

I tried to extract some TODOs from your comments above. Please feel free to correct and extend. I have the idea that my initial post in #327 should be a collection of TODOs acting as roadmap. I gave it an initial structure and I hope you can edit it also. Let me know if there are problems.

**Reply by spwoodcock** [reply]
I'll be back from holidays in a few days and can update #320 shortly 👍
